### PR TITLE
[DT-2269] Don't redirect the /api/user/me call if it fails with a 401 

### PR DESCRIPTION
**Pull request by otchet-broad**
### Addresses

[https://broadworkbench.atlassian.net/browse/DT-2269](https://broadworkbench.atlassian.net/browse/DT-2269)

### Summary

The backend will return a 401 when /api/user/me is called.  The SigninButton.tsx code depends on the logic continuing to flow so we must exclude this call from the traditional 401 handling so that the user account can be created.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/libs/ajax.js [hunk 1]
```diff
@@ -18,7 +18,9 @@ axios.interceptors.response.use(function (response) {
 }, function (error) {
   // Default to a 502 when we can't get a real response object.
   const status = getOr(502)('response.status')(error)
-  if (status === 401) {
+  const url = error.config.url
+  const data = error.config.data
+  if (status === 401 && !(url.endsWith('/api/user/me') && data === undefined)) {
```
Comment (kevinmarete) [inline src/libs/ajax.js]: Nit suggestions..

```suggestion
    const { url, data } = error.config ?? {};
    const isMeCheck = url?.endsWith("/api/user/me") && data === undefined;
    
    if (status === 401 && !isMeCheck) {
      redirectOnLogout();
    }
```

**Review by kevinmarete — APPROVED** [review]

File: src/libs/ajax.js (hunk 1 above)
Comment (rushtong) [reply src/libs/ajax.js]: A bit of history here, the `status === 401` check has caused us grief in the past, especially when integrating with external services ... if they throw a 401, we log the user out incorrectly.

**Review by rushtong — APPROVED** [review]

File: src/libs/ajax.js (hunk 1 above)
Comment (otchetbluehalo) [reply src/libs/ajax.js]: Started doing this refactor after I reviewed the PR for myself.  Take a look at how it ended up.

Comment (otchet-broad) [reply src/libs/ajax.js]: @kevinmarete - I started doing this refactor after I reviewed the PR for myself. Take a look at how it ended up.

**Review by kevinmarete — APPROVED** [review]
Looks good!
